recipe/patches/3.16.2/0001-find_-Add-debug-logging-infrastructure.patch

```diff
@@ -1,478 +0,0 @@
-+  {
-+    std::string debugBuffer;
-+    return this->GetModulesFile(name, system, false, debugBuffer);
-+  }
-+
-+  std::string GetModulesFile(const std::string& name, bool& system, bool debug,
-+                             std::string& debugBuffer) const;
- 
-   //! Set/Get a property of this directory
-   void SetProperty(const std::string& prop, const char* value);
-diff --git a/Source/cmSearchPath.h b/Source/cmSearchPath.h
-index 2a576ed59b..3ecc73bca2 100644
---- a/Source/cmSearchPath.h
-+++ b/Source/cmSearchPath.h
-@@ -5,6 +5,7 @@
- 
- #include "cmConfigure.h" // IWYU pragma: keep
- 
-+#include <cstddef>
- #include <set>
- #include <string>
- #include <vector>
-@@ -27,6 +28,7 @@ public:
-   ~cmSearchPath();
- 
-   const std::vector<std::string>& GetPaths() const { return this->Paths; }
-+  std::size_t size() const { return this->Paths.size(); }
- 
-   void ExtractWithout(const std::set<std::string>& ignore,
-                       std::vector<std::string>& outPaths,
-diff --git a/Source/cmake.cxx b/Source/cmake.cxx
-index f63a264006..6a7511e8c1 100644
---- a/Source/cmake.cxx
-+++ b/Source/cmake.cxx
-@@ -746,6 +746,9 @@ void cmake::SetArgs(const std::vector<std::string>& args)
-         return;
-       }
-       this->SetLogLevel(logLevel);
-+    } else if (arg.find("--debug-find", 0) == 0) {
-+      std::cout << "Running with debug output on for the `find` commands.\n";
-+      this->SetDebugFindOutputOn(true);
-     } else if (arg.find("--trace-expand", 0) == 0) {
-       std::cout << "Running with expanded trace output on.\n";
-       this->SetTrace(true);
-diff --git a/Source/cmake.h b/Source/cmake.h
-index 687c1056b1..ad0ed50968 100644
---- a/Source/cmake.h
-+++ b/Source/cmake.h
-@@ -389,6 +389,10 @@ public:
-   bool GetDebugOutput() { return this->DebugOutput; }
-   void SetDebugOutputOn(bool b) { this->DebugOutput = b; }
- 
-+  //! Do we want debug output from the find commands during the cmake run.
-+  bool GetDebugFindOutput() { return this->DebugFindOutput; }
-+  void SetDebugFindOutputOn(bool b) { this->DebugFindOutput = b; }
-+
-   //! Do we want trace output during the cmake run.
-   bool GetTrace() { return this->Trace; }
-   void SetTrace(bool b) { this->Trace = b; }
-@@ -549,6 +553,7 @@ private:
-   ProgressCallbackType ProgressCallback;
-   WorkingMode CurrentWorkingMode = NORMAL_MODE;
-   bool DebugOutput = false;
-+  bool DebugFindOutput = false;
-   bool Trace = false;
-   bool TraceExpand = false;
-   cmGeneratedFileStream TraceFile;
-diff --git a/Source/cmakemain.cxx b/Source/cmakemain.cxx
-index 6d3e6ee300..04ab27c6f8 100644
---- a/Source/cmakemain.cxx
-+++ b/Source/cmakemain.cxx
-@@ -77,6 +77,7 @@ const char* cmDocumentationOptions[][2] = {
-     "Do not delete the try_compile build tree. Only "
-     "useful on one try_compile at a time." },
-   { "--debug-output", "Put cmake in a debug mode." },
-+  { "--debug-find", "Put cmake find in a debug mode." },
```

Comment:
It seems like this is now present in 3.18
https://gitlab.kitware.com/cmake/cmake/-/blob/master/Source/cmake.cxx#L1112